### ISSUE TITLE
Issue with DDA Quant Ion Level Interface

### ISSUE THREAD

**Issue by KlemensFroehlich** [Proteobench/ProteoBench#179]
hi everyone
I just wanted to start testing the DDA module on the weekend.
I installed the package via my anaconda installation.
I set up a new environment (with Python 3.11.5 as default by my anaconda installation)
downloaded the package from github
installed it using flit as described in the instructions.
I can run the browser application, however when I try to call the DDA Quant Ion Level module I get this error message:

ModuleNotFoundError: No module named 'proteobench'
Traceback:
File "C:\Users\froehl0004\AppData\Local\anaconda3\envs\proteoENV\Lib\site-packages\streamlit\runtime\scriptrunner\script_runner.py", line 534, in _run_script
    exec(code, module.__dict__)
File "C:\pythonTMP\proteobench\webinterface\pages\DDA_Quant.py", line 12, in <module>
    from proteobench.modules.dda_quant.module import Module

Did I do something wrong? Should I use a different python version?

Best, Klemens

**Comment by enryH** [comment]
Hmm. I guess that's the same issue @scaramonche once had. How did you create your new environment?

Could you try to install python and flit into the new environment (latest should be okay):

```bash
conda create -n proteobench python flit
```

This ensure that flit has it's independent python executable and the packages are installed into the correct folder in the Anaconda environment folders. It could be that flit installed all dependencies into your base environment...

**Comment by enryH** [comment]
@mlocardpaulet Otherwise he could use the hosted pre-release version to try it out?

**Comment by mlocardpaulet** [comment]
> @mlocardpaulet Otherwise he could use the hosted pre-release version to try it out?

You mean the one at RUB? That would be ideal indeed. I am just not sure it is ready yet. 
@KlemensFroehlich thanks for trying it out, and sorry to hear that you had issues. Unfortunately, I cannot help. Not sure what went wrong there. Maybe an issue with streamlit version? You could try updating it.
What Henry is referring to is that we plan to have a version available soon without any need for you to install anything. It would just be a url. So you could wait for it.

**Comment by KlemensFroehlich** [comment]
Thank you for your responses

(base) PS C:\Users\froehl0004> conda create -n proteobench python flit
(base) PS C:\Users\froehl0004> conda activate proteobench
(proteobench) PS C:\Users\froehl0004> cd 'C:\pythonTMP\proteobench\'
(proteobench) PS C:\pythonTMP\proteobench> python --version
Python 3.11.5
(proteobench) PS C:\pythonTMP\proteobench> flit install -s

did not solve the problem. Same error message

i noticed that in the powershell there was a Windows error:

OSError: [WinError 1314] A required privilege is not held by the client: 'C:\\pythonTMP\\proteobench\\proteobench' -> 'C:\\Users\\froehl0004\\AppData\\Local\\anaconda3\\envs\\proteobench\\Lib\\site-packages\\proteobench' (

So I tried to run anaconda powershell as admin with a new environment

**now I get a new error:**

ImportError: Failed to initialize: Bad git executable. The git executable must be specified in one of the following ways: - be included in your $PATH - be set via $GIT_PYTHON_GIT_EXECUTABLE - explicitly set via git.refresh() All git commands will error until this is rectified. This initial warning can be silenced or aggravated in the future by setting the $GIT_PYTHON_REFRESH environment variable. Use one of the following values: - quiet|q|silence|s|none|n|0: for no warning or exception - warn|w|warning|1: for a printed warning - error|e|raise|r|2: for a raised exception Example: export GIT_PYTHON_REFRESH=quiet
Traceback:
File "C:\Users\froehl0004\AppData\Local\anaconda3\envs\proteobenchADM\Lib\site-packages\streamlit\runtime\scriptrunner\script_runner.py", line 534, in _run_script
    exec(code, module.__dict__)
File "C:\pythonTMP\proteobench\webinterface\pages\DDA_Quant.py", line 12, in <module>
    from proteobench.modules.dda_quant.module import Module
File "C:\Users\froehl0004\AppData\Local\anaconda3\envs\proteobenchADM\Lib\site-packages\proteobench\modules\dda_quant\module.py", line 15, in <module>
    from proteobench.github.gh import clone_repo, pr_github, read_results_json_repo
File "C:\Users\froehl0004\AppData\Local\anaconda3\envs\proteobenchADM\Lib\site-packages\proteobench\github\gh.py", line 5, in <module>
    from git import Repo
File "C:\Users\froehl0004\AppData\Local\anaconda3\envs\proteobenchADM\Lib\site-packages\git\__init__.py", line 168, in <module>
    raise ImportError("Failed to initialize: {0}".format(_exc)) from _exc

**in powershell the error reads:**

(proteobenchADM) PS C:\pythonTMP\proteobench\webinterface> streamlit run Home.py

  You can now view your Streamlit app in your browser.

  Local URL: http://localhost:8501
  Network URL: http://131.152.98.135:8501

2023-12-18 16:43:17.092 Uncaught app exception
Traceback (most recent call last):
  File "C:\Users\froehl0004\AppData\Local\anaconda3\envs\proteobenchADM\Lib\site-packages\git\__init__.py", line 166, in <module>
    refresh()
  File "C:\Users\froehl0004\AppData\Local\anaconda3\envs\proteobenchADM\Lib\site-packages\git\__init__.py", line 153, in refresh
    if not Git.refresh(path=path):
           ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\froehl0004\AppData\Local\anaconda3\envs\proteobenchADM\Lib\site-packages\git\cmd.py", line 391, in refresh
    raise ImportError(err)
ImportError: Bad git executable.
The git executable must be specified in one of the following ways:
    - be included in your $PATH
    - be set via $GIT_PYTHON_GIT_EXECUTABLE
    - explicitly set via git.refresh()

All git commands will error until this is rectified.

This initial warning can be silenced or aggravated in the future by setting the
$GIT_PYTHON_REFRESH environment variable. Use one of the following values:
    - quiet|q|silence|s|none|n|0: for no warning or exception
    - warn|w|warning|1: for a printed warning
    - error|e|raise|r|2: for a raised exception

Example:
    export GIT_PYTHON_REFRESH=quiet


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\froehl0004\AppData\Local\anaconda3\envs\proteobenchADM\Lib\site-packages\streamlit\runtime\scriptrunner\script_runner.py", line 534, in _run_script
    exec(code, module.__dict__)
  File "C:\pythonTMP\proteobench\webinterface\pages\DDA_Quant.py", line 12, in <module>
    from proteobench.modules.dda_quant.module import Module
  File "C:\Users\froehl0004\AppData\Local\anaconda3\envs\proteobenchADM\Lib\site-packages\proteobench\modules\dda_quant\module.py", line 15, in <module>
    from proteobench.github.gh import clone_repo, pr_github, read_results_json_repo
  File "C:\Users\froehl0004\AppData\Local\anaconda3\envs\proteobenchADM\Lib\site-packages\proteobench\github\gh.py", line 5, in <module>
    from git import Repo
  File "C:\Users\froehl0004\AppData\Local\anaconda3\envs\proteobenchADM\Lib\site-packages\git\__init__.py", line 168, in <module>
    raise ImportError("Failed to initialize: {0}".format(_exc)) from _exc
ImportError: Failed to initialize: Bad git executable.
The git executable must be specified in one of the following ways:
    - be included in your $PATH
    - be set via $GIT_PYTHON_GIT_EXECUTABLE
    - explicitly set via git.refresh()

All git commands will error until this is rectified.

This initial warning can be silenced or aggravated in the future by setting the
$GIT_PYTHON_REFRESH environment variable. Use one of the following values:
    - quiet|q|silence|s|none|n|0: for no warning or exception
    - warn|w|warning|1: for a printed warning
    - error|e|raise|r|2: for a raised exception

Example:
    export GIT_PYTHON_REFRESH=quiet

2023-12-18 16:43:58.526 Uncaught app exception
Traceback (most recent call last):
  File "C:\Users\froehl0004\AppData\Local\anaconda3\envs\proteobenchADM\Lib\site-packages\git\__init__.py", line 166, in <module>
    refresh()
  File "C:\Users\froehl0004\AppData\Local\anaconda3\envs\proteobenchADM\Lib\site-packages\git\__init__.py", line 153, in refresh
    if not Git.refresh(path=path):
           ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\froehl0004\AppData\Local\anaconda3\envs\proteobenchADM\Lib\site-packages\git\cmd.py", line 391, in refresh
    raise ImportError(err)
ImportError: Bad git executable.
The git executable must be specified in one of the following ways:
    - be included in your $PATH
    - be set via $GIT_PYTHON_GIT_EXECUTABLE
    - explicitly set via git.refresh()

All git commands will error until this is rectified.

This initial warning can be silenced or aggravated in the future by setting the
$GIT_PYTHON_REFRESH environment variable. Use one of the following values:
    - quiet|q|silence|s|none|n|0: for no warning or exception
    - warn|w|warning|1: for a printed warning
    - error|e|raise|r|2: for a raised exception

Example:
    export GIT_PYTHON_REFRESH=quiet


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\froehl0004\AppData\Local\anaconda3\envs\proteobenchADM\Lib\site-packages\streamlit\runtime\scriptrunner\script_runner.py", line 534, in _run_script
    exec(code, module.__dict__)
  File "C:\pythonTMP\proteobench\webinterface\pages\DDA_Quant.py", line 12, in <module>
    from proteobench.modules.dda_quant.module import Module
  File "C:\Users\froehl0004\AppData\Local\anaconda3\envs\proteobenchADM\Lib\site-packages\proteobench\modules\dda_quant\module.py", line 15, in <module>
    from proteobench.github.gh import clone_repo, pr_github, read_results_json_repo
  File "C:\Users\froehl0004\AppData\Local\anaconda3\envs\proteobenchADM\Lib\site-packages\proteobench\github\gh.py", line 5, in <module>
    from git import Repo
  File "C:\Users\froehl0004\AppData\Local\anaconda3\envs\proteobenchADM\Lib\site-packages\git\__init__.py", line 168, in <module>
    raise ImportError("Failed to initialize: {0}".format(_exc)) from _exc
ImportError: Failed to initialize: Bad git executable.
The git executable must be specified in one of the following ways:
    - be included in your $PATH
    - be set via $GIT_PYTHON_GIT_EXECUTABLE
    - explicitly set via git.refresh()

All git commands will error until this is rectified.

This initial warning can be silenced or aggravated in the future by setting the
$GIT_PYTHON_REFRESH environment variable. Use one of the following values:
    - quiet|q|silence|s|none|n|0: for no warning or exception
    - warn|w|warning|1: for a printed warning
    - error|e|raise|r|2: for a raised exception

when I use pip to install the proteobench package, I get the same error.

Best, Klemens

**Comment by KlemensFroehlich** [comment]
ah and I am using streamlit 1.29.0, which I think is the latest streamlit version according to github
https://github.com/streamlit/streamlit

Best, Klemens

**Comment by enryH** [comment]
Do you have a system wide installation of git? I guess you cloned the repo using git? (`git status` should then work). I had to manually install [GitPython ](https://github.com/gitpython-developers/GitPython )once, but it should be installed using flit. 

Just to check: Did you execute `flit install -s` in the powershell as admin?

**Comment by KlemensFroehlich** [comment]
I think I got it:
I ran anaconda powershell as admin
I installed git and flit globally
I created a new env with 
```bash
conda create proteobench python flit git
conda actviate proteobench
```
Don't know whether this is redundant as I already had the global git and flit installations :)

then clondes the git repository:

```bash
git clone https://github.com/Proteobench/ProteoBench
```

then I used flit for the package installation:

```bash
flit install -s
```

and followed the instructions for the localhost web browser interface. Now the example for the DDA module works.

Thanks for your hel @enryH 
I think it was either that I downloaded the repository before instead of cloning or it was a writing privilege issue coming from windows.

Will start testing different FragPipe stuff now and try to break the tool ;)

Best, Klemens
